libusb/libusbi.h

```diff
@@ -257,7 +257,14 @@ static inline void list_splice_front(struct list_head *list, struct list_head *h
 
 static inline void *usbi_reallocf(void *ptr, size_t size)
 {
-	void *ret = realloc(ptr, size);
+	void *ret;
+
+	if (size == 0) {
```

Comment:
Hmmm, I'm not convinced about the semantics you describe.  It doesn't match FreeBSD for example:

https://github.com/freebsd/freebsd-src/blob/main/lib/libc/stdlib/reallocf.c

would doing exactly what FreeBSD does pacify your tool?

Reply:
Likely yes. If you prefer that, then I would say pull in the code and see if anyone complains about it in the future.

Reply:
Hmm, I just looked at it, and was about to change it. And then, I had this distinct feeling that the freebsd code would still trigger that warning.

Not sure, I don't really care. My take is, please just drop the patch if you don't like it as-is and put in the alternative version.

Reply:
I also note that C23 has now clarified that passing 0 as size to `realloc` is undefined behaviour, see https://en.cppreference.com/w/c/memory/realloc

But of course `reallocf` is not part of C, it's from FreeBSD (and also in macOS). Which is why I think we'd be much better off exactly copying FreeBSD's implementation.

I can do that in a separate PR if you want to just drop this commit from this PR (which already has merge conflicts apparently).

Reply:
Hey, wait, `usbi_reallocf` is not public API, and nothing in libusb even uses it.  Maybe we can just remove it...

Reply:
Hah, even better then, I suppose :-)